js/player.js

```diff
@@ -150,6 +151,24 @@ function($, playlist, ytIframe){
             }
         }
     }
+    var clearPlaylist = function(){
+      //clear playlist
+      playlist = [];
+      //clear total duration
+      totalDuration = 0;
+      //removes all the videos
+      $('#playlist li').remove();
+      //save the playlist
+      localStorage.setItem('playlist', JSON.stringify(playlist));
+      localStorage.setItem('new_playlist_saved', JSON.stringify(true));
+      //reset duration
+      localStorage.setItem('total_duration', JSON.stringify(totalDuration));
+      //reload page
+      location.reload();
+      //change title
+      document.title= "Youtube Fast Playlist";
+    }
```

Comment:
This whole function doesn't make sense here, because it makes changes to the playlist. It should be placed in `playlist.js`. And why is `location.reload()` necessary?

Reply:
Thought reloading page will make it more cleaner. I will look on how to do the same without reloading the page.

Reply:
Another suggestion is that, when you set `playlist = []` and `total_duration = 0`, calling `savePlaylist()`  and  `saveDuration()` will be cleaner instead of directly setting it in `localStorage`.

Reply:
Okay.